### PR TITLE
EVSS Migration - 526 Submit - Transform Service - Banking Info

### PR DESCRIPTION
**Pull request by sethdarragile6**
## Summary

- This PR implements the transformation of the Lighthouse request "DirectDeposit" object
- Disability Benefits Experience Team 1 (DBex TRex) 🦖
- No flipper needed, as it will not be called until further down the road

## Related issue(s)
- [61552](https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/61552)

## Testing done

- Tests run successfully, they are simply validating the infrastructure (`bundle exec rspec spec/lib/evss/disability_compensation_form/form526_to_lighthouse_transform_spec.rb`)

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Review by navaeddie — APPROVED** [review]
neat and tidy

File: lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb [hunk 1]
```diff
@@ -253,6 +255,21 @@ def transform_secondary_disabilities(disability_source)
           sd
         end
       end
+
+      def transform_direct_deposit(direct_deposit_source)
+        direct_deposit = Requests::DirectDeposit.new
+        if direct_deposit_source.nil?
```
Comment (aurora-a-k-a-lightning) [inline lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb:261]: for this use case - missing a direct deposit node in the EVSS request structure - we agreed to **_not_** send this `'directDeposit'`node for now until the business logic is clear from stakeholders!
let's remove this node from the LH request body if there is no `directDeposit` node

File: lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb [hunk 2]
```diff
@@ -25,6 +25,8 @@ def transform(evss_data)
         disabilities = form526['disabilities']
         lh_request_body.disabilities = transform_disabilities(disabilities)
 
+        direct_deposit = form526['directDeposit']
+        lh_request_body.direct_deposit = transform_direct_deposit(direct_deposit)
```
Comment (aurora-a-k-a-lightning) [inline lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb:29]: for the missing `directDeposit` node, we can place the `if direct_deposit.present?` at the end of this line to check before adding to the request body

File: spec/lib/evss/disability_compensation_form/form526_to_lighthouse_transform_spec.rb [hunk 3]
```diff
@@ -153,4 +154,24 @@
...
+      result = transformer.send(:transform_direct_deposit, data)
+      expect(result.financial_institution_name).to eq('SomeBank')
+      expect(result.account_type).to eq('CHECKING')
+      expect(result.account_number).to eq('123123123123')
+      expect(result.routing_number).to eq('123123123')
+      expect(result.no_account).to eq(false)
+    end
+
+    it 'sends no_account = true if none exists' do
+      result = transformer.send(:transform_direct_deposit, nil)
```
Comment (aurora-a-k-a-lightning) [inline spec/lib/evss/disability_compensation_form/form526_to_lighthouse_transform_spec.rb:172]: this test will have to change :yum:

**Review by aurora-a-k-a-lightning — CHANGES_REQUESTED** [review]
let's keep our same business logic until told what to do!

File: lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb (hunk 1 above)
Comment (sethdarragile6) [reply lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb:261]: PIm1apF - Imgur

Comment (sethdarragile6) [reply lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb]: @aurora-a-k-a-lightning this should be fixed now. my bad!

File: lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb (hunk 2 above)
Comment (sethdarragile6) [reply lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb]: @aurora-a-k-a-lightning fixed

**Review by navaeddie — APPROVED** [review]
still LGTM

**Review by aurora-a-k-a-lightning — APPROVED** [review]

**Review by holdenhinkle — APPROVED** [review]
Very nice!

**Pull request merged**